### PR TITLE
Check that libdl is enabled for profiling

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: cmake/kokkos_tpls.cmake [hunk 1]
```diff
@@ -15,6 +15,10 @@ KOKKOS_TPL_OPTION(CUDA    Off)
 KOKKOS_TPL_OPTION(LIBRT   Off)
 KOKKOS_TPL_OPTION(LIBDL   On)
 
+IF(KOKKOS_ENABLE_PROFILING AND NOT KOKKOS_ENABLE_LIBDL)
+  MESSAGE(SEND_ERROR "Kokkos_ENABLE_PROFILING requires Kokkos_ENABLE_LIBDL=ON")
```
Comment (dalg24) [inline cmake/kokkos_tpls.cmake]: Did you consider emitting a warning and disabling the profiling hooks if libdl is not found?

**Comment by dalg24** [comment]
Retest this please

File: cmake/kokkos_tpls.cmake (hunk 1 above)
Comment (masterleinad) [reply cmake/kokkos_tpls.cmake]: I thought about changing the default value, but this file is parsed after the one setting up the profiling. 
I rather have a configure time error than overriding what the user requests (even emitting a warning).

Comment (dalg24) [reply cmake/kokkos_tpls.cmake]: I am not 100% sold on that behavior but I won't object if we issue a better error message.

Comment (masterleinad) [reply cmake/kokkos_tpls.cmake]: What don't you like about the error message? In which regard could it be improved?

Comment (dalg24) [reply cmake/kokkos_tpls.cmake]: On second thought, this is fine.  I didn't realize the current behavior is to error out if libdl is not found, even for the case where the user did not explicitly enabled it.

**Comment by dalg24** [comment]
Any other opinion on whether to disable profiling and emit a warning or error out as proposed here?

**Review by dalg24 — APPROVED** [review]

**Pull request merged**